connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -124,7 +127,22 @@ public void poll(long timeout) {
 
         do {
             if (coordinatorUnknown()) {
-                ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+                log.debug("Broker coordinator is marked unknown. Attempting discovery with a timeout of {}ms",
+                        coordinatorDiscoveryTimeoutMs);
+                if (ensureCoordinatorReady(time.timer(coordinatorDiscoveryTimeoutMs))) {
+                    log.debug("Broker coordinator is ready");
+                } else {
+                    log.debug("Can not connect to broker coordinator");
+                    if (assignmentSnapshot != null && !assignmentSnapshot.failed()) {
+                        log.info("Broker coordinator was unreachable for {}ms. Revoking previous assignment {} to " +
+                                "avoid running tasks while not being a member the group", coordinatorDiscoveryTimeoutMs, assignmentSnapshot);
+                        listener.onRevoked(assignmentSnapshot.leader(), assignmentSnapshot.connectors(), assignmentSnapshot.tasks());
+                        assignmentSnapshot.connectors().clear();
+                        assignmentSnapshot.tasks().clear();
+                        assignmentSnapshot.revokedConnectors().clear();
+                        assignmentSnapshot.revokedTasks().clear();
```

Comment:
Will all of these methods to clear state within the assignment snapshot appear to happen atomically w/r/t when the assignment snapshot is read elsewhere in this class?

Reply:
Good point. We don't lock access to this collection. Setting to `null` to avoid any concurrent access issues (from the threads that export metrics potentially).